modules/product/src/Form/ProductTypeForm.php

```diff
@@ -100,6 +100,8 @@ public function form(array $form, FormStateInterface $form_state) {
       '#default_value' => $product_type->getVariationType(),
       '#options' => $variation_types,
       '#required' => TRUE,
+      '#description' => $this->t('This value cannot be changed once you have saved this product type.'),
```

Comment:
We don't provide such a description for other similar cases (such as machine names), so this might be unneeded?